pkg/apis/discovery/v1alpha1/clusterducktype_types.go

```diff
@@ -71,6 +72,10 @@ type ClusterDuckTypeSpec struct {
 	// Selectors is a list of selectors for CustomResourceDefinitions to identify a duck type.
 	// +optional
 	Selectors []CustomResourceDefinitionSelector `json:"selectors,omitempty"`
+
+	// ClusterRoleRef is a reference to the Aggregating ClusterRole used by the duck type to manage the ducks
+	// +optional
+	ClusterRoleRef *rbacv1.RoleRef `json:"clusterRoleRef,omitempty"`
```

Comment:
I think this is a great start. 

I do think letting people specify the label selector in the future is going to be useful but we can punt on it in this first pass. But  to make room, I would ask you make this change:

from 

```
	ClusterRoleRef *rbacv1.RoleRef `json:"clusterRoleRef,omitempty"`
```

to

```
	Role Role`json:"role,omitempty"`
...
type Role struct {
   Ref *rbacv1.RoleRef `json:"ref,omitempty"`
}
```

in use

```
role:
    ref:
        kind: ClusterRole
        name: channelable-manipulator
        apiGroup: rbac.authorization.k8s.io
```

The fact that it is cluster or namespace comes from the kind, we can choose to only accept `ClusterRole` in the webhook, and that gives us a way to add DuckType (namespaced) without changing the spec shape a ton.


Then, when this all lands, we can add the label selector to use as an override:
...
type Role struct {
   Ref *rbacv1.RoleRef `json:"ref,omitempty"`
   LabelSelector string `json:"selector"`
}
```

or something like this... as a followup.

Thoughts? I think we are close, I like the resulting status on this new update.

Reply:
Makes sense based on the earlier discussion 👍